src/models/user.ts

```diff
@@ -265,34 +276,41 @@ export const Users: UserCollection = {
     return { ...res, token } as SingleUseToken;
   },
 
-  async fromPasswordResetToken (client, token) {
-    // Find token
-    // // No token -> Error
-    const tokens = await client('single_use_token')
-      .select()
-      .where({ deleted_at: null, used_at: null })
-      .andWhere('expires_at', '>', new Date());
-
-    const tokenRow = await findAsync(tokens, (a: any) => compare(token, a.token));
-
-    if (!tokenRow) {
-      throw new Error('Unrecognised reset token');
-    }
-
-    // Find associated user
-    // // No associated user -> Error
-    const [{ user_account_id: id }] = await client('user_secret_reset')
-      .select('user_account_id')
-      .where({ single_use_token_id: tokenRow.single_use_token_id });
-
-    if (!id) {
-      throw new Error('No associated user');
-    }
-
+  async usePasswordResetToken (client, email, token) {
+    // get user
+    const user = await Users.getOne(client, { where: { email } });
+    if (!user) throw new Error('E-mail not recognised');
+
+    // get valid token matching user
+    const [match] = await client('single_use_token')
+      .select('*')
+      .innerJoin(
+        'user_secret_reset',
+        'user_secret_reset.single_use_token_id',
+        'single_use_token.single_use_token_id')
+      .innerJoin(
+        'user_account',
+        'user_account.user_account_id',
+        'user_secret_reset.user_account_id')
+      .where({
+        'user_account.email': email,
+        'single_use_token.used_at': null,
+        'single_use_token.deleted_at': null,
+      })
+      .andWhereRaw('single_use_token.expires_at > ?', [moment().toISOString()]);
+
+    if (!match) throw new Error('Token not recognised');
+
+    // check token hash matches supplied token
+    const isValid = await compare(token, match.token);
+
+    if (!isValid) throw new Error('Token not recognised');
+
+    // update token row as used
     await client('single_use_token')
-      .update({ used_at: new Date() })
-      .where({ single_use_token_id: tokenRow.single_use_token_id });
+      .update({ used_at: moment().toISOString() })
```

Comment:
No action, but is there a reason for `moment` here? There's no real difference between `new Date()` and `moment()` if they're only being used for the ISO86001 output. 🤷🏼‍♂️